src/tournament_commands/player_commands/decklist.rs

```diff
@@ -5,5 +5,35 @@ use serenity::prelude::*;
 #[command("decklist")]
 #[description("Prints out one of your decklists.")]
 async fn decklist(ctx: &Context, msg: &Message, _: Args) -> CommandResult {
-    todo!()
+     /* Get references to needed data from context */
+     let data = ctx.data.read().await;
+     let name_and_id = data.get::<TournamentNameAndIDMapContainer>().unwrap();
+     let gld_tourns = data.get::<GuildAndTournamentIDMapContainer>().unwrap();
+     let player_name = msg.author.id.0.to_string();
+
+     let id_iter = gld_tourns.get_left_iter(&msg.guild_id.unwrap()).unwrap();
+     let tourn_id = match id_iter.len() {
```

Comment:
Now, `id_iter` will only yield the ids of tournaments that the player is registered in. We just need to change the message in the `0` branch and add some `args` checks in the `_` branch.